src/switch_core_session.c

```diff
@@ -1459,6 +1459,78 @@ SWITCH_DECLARE(uint32_t) switch_core_session_flush_and_publish_private_events(sw
 	return x;
 }
 
+SWITCH_DECLARE(uint32_t) switch_core_session_flush_and_publish_private_media_events(switch_core_session_t *session)
+{
+	switch_status_t status = SWITCH_STATUS_FALSE;
+	switch_queue_t *queue_pri, *queue;
+	int x = 0;
+	void *pop;
+	
+	if (session->private_event_queue) {
+		switch_memory_pool_t *pool = switch_core_session_get_pool(session);
+		switch_queue_create(&queue_pri, SWITCH_EVENT_QUEUE_LEN, pool);
```

Comment:
I would recommend to only create the queue if needed. If for example the private_event_queue_pri or private_event_queue is empty then will still always create the queue and replace the existing queue.